src/cogent3/core/new_moltype.py

```diff
@@ -532,32 +532,141 @@ def _(self, seq: bytes) -> bool:
 
     @is_gapped.register
     def _(self, seq: numpy.ndarray) -> bool:
-        return (seq == self.degen_gapped_alphabet.gap_index).any()
+        gaps_indices = [self.degen_gapped_alphabet.index(gap) for gap in self.gaps]
+        return numpy.isin(seq, gaps_indices).any()
 
     @functools.singledispatchmethod
-    def get_degenerate_positions(self, seq, include_gap=True) -> numpy.ndarray:
-        """returns a boolean array indicating degenerate positions"""
+    def get_degenerate_positions(self, seq, include_gap: bool = True) -> list[int]:
+        """returns indices matching degenerate characters"""
         raise TypeError(f"{type(seq)} not supported")
 
     @get_degenerate_positions.register
-    def _(self, seq: numpy.ndarray, include_gap=True) -> numpy.ndarray:
+    def _(self, seq: numpy.ndarray, include_gap: bool = True) -> list[int]:
```

Comment:
you can remove the singledispatch decoration. Juts convert to using this method with a small tweak
```python
seq = self.degen_gapped_alphabet.to_indices(seq)
...
```
The alphabet.to_indices(seq) method handles all the different types!